[DiscordRPC] Fix non-functioning global toggle

bagusnl opened the pull request:
# Main Goal
Fix global RPC toggle not actually disabling RPC.
Also disable RPC regional setting when global toggle is set to disabled.

Closes #806 

## PR Status :
- Overall Status : Done
- Commits : Done
- Synced to base (Collapse:main) : Yes
- Build status : OK
- Crashing : No
- Bug found caused by PR : 0

### Templates

<details>
  <summary>Changelog Prefixes</summary>
  
  ```
    **[New]**
    **[Imp]**
    **[Fix]**
    **[Loc]**
    **[Doc]**
  ```

</details>

bagusnl committed:
[DiscordRPC] Fix toggle not actually disabling RPC

Signed-off-by: Bagus Nur Listiyono <dzakibagus@gmail.com>

bagusnl committed:
[DiscordRPC] Disable regional QS toggle if global toggle is off

Signed-off-by: Bagus Nur Listiyono <dzakibagus@gmail.com>

neon-nyan reviewed (changes requested):
I think all should be good except for those request changes below.

neon-nyan committed:
Simplify IsRpcEnabled

Use direct assignment to field and use implicit cast instead of ToBoolNullable() as the boolean will return false as default value afterwards.

bagusnl commented on CollapseLauncher/Classes/DiscordPresence/DiscordPresenceManager.cs:
```diff
@@ -37,6 +37,24 @@ public enum ActivityType
     {
         #region Properties
+
+        [field: NonSerialized]
```
I literally don't know what went over my head when creating the field, I guess my brain were just too trained to use these primitives field/var caching methods xd.

Thanks for the fix tho!
